### PR TITLE
DataGrid.Pdf - Indents for grouped rows with summaries

### PR DESCRIPTION
**Pull request by KuznetsovVN**

### CODE REVIEW

**Review by novsstation — DISMISSED** [review]

**Review by EugeniyKiyashko — DISMISSED** [review]

**Review by novsstation — DISMISSED** [review]

File: js/exporter/jspdf/export_data_grid_2.js [hunk 1]
```diff
@@ -48,7 +48,28 @@ function exportDataGrid(doc, dataGrid, options) {
                             pdfCell.colSpan = cellMerging.colspan;
                         }
                     } else if(rowType === 'group') {
-                        pdfCell.colSpan = columns.length - 1;
+                        if(cellData.cellSourceData.groupSummaryItems?.length > 0) {
+                            pdfCell._summaryItem = true;
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js]: '_summaryItem' looks strange: there is no 'summary' in 'pdfCell'
Can we rename it? 'isEmpty' ?

File: js/exporter/jspdf/export_data_grid_2.js [hunk 2]
```diff
@@ -48,7 +48,28 @@ function exportDataGrid(doc, dataGrid, options) {
...
+                        }
+
+                        if(cellIndex > 0) {
+                            const prevCellsHaveSummaryItems = currentRow.slice(1).filter(pdfCell => pdfCell._summaryItem).length > 0;
+                            if(!pdfCell._summaryItem && !prevCellsHaveSummaryItems) {
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid_2.js]: '_summaryItem' is used in this method only but declared in a long living object.
Can we change it to a local variable?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/export_data_grid_2.js (hunk 2 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid_2.js]: Ok, we can remove the '_summaryItem' field and use the 'isDefined(pdfCell.text)' expression instead. In this case we will assume that if the text is defined in pdfCell of the group row, then this is a summary item aligned by column. Except the first cell.

Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid_2.js]: Done

**Review by IgnatovDan — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**